frontend/src/libs/components/order-card/order-card.tsx

```diff
@@ -1,70 +1,73 @@
 import { PlainSvgIconName } from '~/libs/enums/enums.js';
-import { getValidClassNames, pluralizeString } from '~/libs/helpers/helpers.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { actions as orderActions } from '~/slices/orders/order.js';
+import { selectOrder, selectOrderData } from '~/slices/orders/selectors.js';
 
 import { PlainSvgIcon } from '../plain-svg-icon/plain-svg-icon.js';
-import {
-  type DriverProfileInfo,
-  type OrderCurrentStatus,
-  type OrderedTruckInfo,
-  type OrderInitialStatus,
-} from './libs/types/types.js';
+import { CardHeader } from './libs/components/components.js';
 import styles from './styles.module.scss';
 
 type Properties = {
-  driver: DriverProfileInfo;
-  truck: OrderedTruckInfo;
-  initialStatus: OrderInitialStatus;
-  currentStatus: OrderCurrentStatus;
+  className?: string;
+  isDriverShown?: boolean;
 };
 
 const OrderCard: React.FC<Properties> = ({
-  driver: { firstName, lastName, profileURL },
-  truck: { licensePlate },
-  initialStatus: { startLocation, endLocation },
-  currentStatus: { timespanLastUpdated, location, distanceLeft, timespanLeft },
+  className,
+  isDriverShown = true,
 }: Properties) => {
-  const pluralizedKilometers = pluralizeString('km', distanceLeft);
+  const order = useAppSelector(selectOrder);
+  const {
+    origin: startLocation = '',
+    destination: endLocation = '',
+    distanceAndDuration,
+  } = useAppSelector(selectOrderData);
+
+  const { distance, duration } = distanceAndDuration ?? {};
+  const { text: distanceLeft = '' } = distance ?? {};
+  const { text: timespanLeft = '' } = duration ?? {};
+  const { shift, price } = order ?? {};
+  const { truck, driver } = shift ?? {};
+  const { licensePlateNumber: licensePlate = '' } = truck ?? {};
+  const { firstName: firstName = '', lastName: lastName = '' } = driver ?? {};
```

Comment:
So all this stuff could be null or undefined here? I guess if we are showing order-card - then this data should be present